### DISCUSSION TITLE
Support for wireguard-go

### DISCUSSION THREAD

**Discussion by varbhat** [anacrolix/torrent#648] [Ideas]
Please see https://news.ycombinator.com/item?id=28303896 .

**Comment by anacrolix** [comment]
Consider funnelling further discussion here so I can unpack whether it's something that could/should be supported directly in this repo.

**Comment by varbhat** [comment]
@anacrolix ,

I was suggested to integrate "wireguard"(wireguard-go) within exatorrent because it would be nice to have a single binary that can run torrent traffic through a VPN(wireguard).

Since the support must come from anacrolix/torrent library, what do you think of this?

**Comment by anacrolix** [comment]
Yes it could be done by implementing custom Dialers and Listeners I expect.
